src/cljam/algo/pileup.clj

```diff
@@ -142,6 +115,159 @@
          (partial filterv)
          (update p 1))))
 
+(defn- merge-corrected-quals
+  "Merge corrected quals with the uncorrected part."
+  [^SAMAlignment aln ^long correct-start corrected-quals]
+  (let [quals (:quals-at-ref aln)
+        start (.pos aln)
+        len   (count corrected-quals)]
+    (if (<= start correct-start)
+      (into (into (vec (take (- correct-start start) quals)) corrected-quals)
```

Comment:
`subvec` may be more efficient for splitting vectors.

```clojure
      (-> quals
          (subvec 0 (- correct-start start))
          (into corrected-quals)
          (into (subvec quals (+ (- correct-start start) len))))
      (into corrected-quals (subvec len quals)))))
```

Reply:
It is more straightforward. Fixed.